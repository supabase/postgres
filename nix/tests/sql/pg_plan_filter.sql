begin;
  load 'plan_filter';

  create schema v;

  -- create a sample table
  create table v.test_table (
    id serial primary key,
    data text
  );

  -- insert some test data
  insert into v.test_table (data)
  values ('sample1'), ('sample2'), ('sample3');

  set local plan_filter.statement_cost_limit = 0.001;

  select * from v.test_table;

rollback;


