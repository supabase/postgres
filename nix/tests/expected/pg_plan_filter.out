begin;
  load 'plan_filter';
  create schema v;
  -- create a sample table
  create table v.test_table (
    id serial primary key,
    data text
  );
  -- insert some test data
  insert into v.test_table (data)
  values ('sample1'), ('sample2'), ('sample3');
  set local plan_filter.statement_cost_limit = 0.001;
  select * from v.test_table;
ERROR:  plan cost limit exceeded
HINT:  The plan for your query shows that it would probably have an excessive run time. This may be due to a logic error in the SQL, or it maybe just a very costly query. Rewrite your query or increase the configuration parameter "plan_filter.statement_cost_limit".
rollback;
